sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosResponseDiagnosticsTest.java

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.OperationType;
+import com.azure.cosmos.rx.TestSuiteBase;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+public class CosmosResponseDiagnosticsTest extends TestSuiteBase {
+    private CosmosClient gatewayClient;
+    private CosmosDatabase createdDatabase;
+    private CosmosContainer cosmosContainer;
+    private CosmosClient directClient;
+
+    private final static String DATABASE_NAME = "TestDiagnosticDB";
+    private final static String CONTAINER_NAME = "TestDiagnosticContainer";
```

Comment:
please try to use the shared database and collection to avoid master resource throttling.

See DocumentCrudTest as example

Reply:
Done